torch/testing/_comparison.py

```diff
@@ -771,42 +773,59 @@ def _compare_sparse_coo_values(
             identifier="Sparse COO values",
         )
 
-    def _compare_sparse_csr_values(
+    def _compare_sparse_compressed_values(
         self, actual: torch.Tensor, expected: torch.Tensor, *, rtol: float, atol: float, equal_nan: bool
     ) -> None:
-        """Compares sparse CSR tensors by comparing
+        """Compares sparse compressed tensors by comparing
 
         - the number of non-zero elements (nnz) for equality,
-        - the col_indices for equality,
-        - the crow_indices for equality, and
+        - the plain indices for equality,
+        - the compressed indices for equality, and
         - the values for closeness.
         """
+        format_name = {
+            torch.sparse_csr: 'CSR',
+            torch.sparse_csc: 'CSC',
+            torch.sparse_bsr: 'BSR',
+            torch.sparse_bsc: 'BSC'}[actual.layout]
```

Comment:
Can we maybe merge the three dictionaries into one, i.e.

```py
{layout: (format_name, compressed_indices_mth, plain_indices_mth)}
```

Reply:
sure

Reply:
Done.